src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BranchDiscoveryTrait.java

```diff
@@ -245,6 +245,12 @@ public boolean isExcluded(@NonNull SCMSourceRequest request, @NonNull SCMHead he
                     BitbucketRepository source = pullRequest.getSource().getRepository();
                     if (fullName.equalsIgnoreCase(source.getFullName())
                             && pullRequest.getSource().getBranch().getName().equals(head.getName())) {
+                        // End the format with newline to avoid logging this
+                        // result blocked together with a later indexed branch
+                        request.listener().getLogger().format(
+                                "Ignoring %s because current strategy excludes branches "
+                                + "that ARE also filed as a pull request%n"
+                                , head.toString());
```

Comment:
As you report in #284 the output log seems to be
```
Checking branch featureimage/new-nut-configurator
Ignoring SCMHead{'featureimage/new-nut-configurator'} because current strategy excludes branches that ARE also filed as a pull request
```
Instead of head.toString() could you use a method to produce a readable the branch name as in the above log `branch featureimage/new-nut-configurator` ?